packages/npm-mongo/wrapper.js

```diff
@@ -1,6 +1,29 @@
-const useLegacyMongo = !!Package['npm-mongo-legacy'];
+const { MongoClient, MongoServerSelectionError, MongoCompatibilityError } = Npm.require('mongodb');
 
+function connect(client) {
+  return client.connect()
+    .catch(error => {
+      if (error.cause instanceof MongoCompatibilityError && error.message.includes('maximum wire version')) {
+      console.warn(`[DEPRECATION] Legacy MongoDB version detected, using mongo-legacy package: ${error.message}
+        Warning: MongoDB versions <= 3.6 are deprecated. Some Meteor features may not work properly with this version.
+        It is recommended to use MongoDB >= 4.`);
+      if (!Package['npm-mongo-legacy']) {
+        throw new Error('Please, install npm-mongo-legacy package to use this version of MongoDB running "meteor add npm-mongo-legacy"');
+      }
+      return false
+    } else throw new Error(`Failed to initialize Meteor's npm-mongo package: ${error}`);
+  })
+}
+
+connect(new MongoClient(process.env.MONGO_URL)).then(client => {
```

Comment:
This should be done only when process.env.MONGO_URL is set, likely on a prod-like environment. Locally, when using the built-in mongo should not be triggering, it will try to connect on an empty url connection. This is likely the reason all tests fail on this branch.